source/agora/test/GenesisBlock.d

```diff
@@ -16,83 +17,37 @@ module agora.test.GenesisBlock;
 version (unittest):
 
 import agora.api.Validator;
-import agora.common.Amount;
-import agora.common.BitField;
-import agora.common.crypto.Key;
 import agora.common.Hash;
 import agora.consensus.data.Block;
-import agora.consensus.data.Enrollment;
 import agora.consensus.data.Transaction;
 import agora.consensus.Genesis;
 import agora.test.Base;
 
-import std.exception : assumeUnique;
-import std.typecons : tuple;
-
-/// create our own genesis block that uses the given keypair,
-/// returns tuple(block, genesis_tx)
-private auto makeCustomGenesisBlock (in KeyPair key_pair)
-{
-    Transaction GenTx =
-    {
-        TxType.Payment,
-        inputs: [ Input.init ],
-        outputs: [
-            Output(Amount(62_500_000L * 10_000_000L), key_pair.address),
-            Output(Amount(62_500_000L * 10_000_000L), key_pair.address),
-            Output(Amount(62_500_000L * 10_000_000L), key_pair.address),
-            Output(Amount(62_500_000L * 10_000_000L), key_pair.address),
-            Output(Amount(62_500_000L * 10_000_000L), key_pair.address),
-            Output(Amount(62_500_000L * 10_000_000L), key_pair.address),
-            Output(Amount(62_500_000L * 10_000_000L), key_pair.address),
-            Output(Amount(62_500_000L * 10_000_000L), key_pair.address),
-        ],
-    };
-
-    Transaction[] txs = [GenTx];
-    Hash[] merkle_tree;
-    auto merkle_root = Block.buildMerkleTree(txs, merkle_tree);
-
-    immutable(BlockHeader) makeHeader ()
-    {
-        return immutable(BlockHeader)(
-            Hash.init,   // prev
-            0,           // height
-            merkle_root,
-            BitField!uint.init,
-            Signature.init,
-            null,        // enrollments
-        );
-    }
-
-    return tuple(immutable(Block)(
-        makeHeader(),
-        txs.assumeUnique,
-        merkle_tree.assumeUnique
-    ), GenTx);
-}
-
 /// ditto
 unittest
 {
-    import core.thread;
-
-    auto kp = KeyPair.random();
-    auto genesis = makeCustomGenesisBlock(kp);
-
-    TestConf conf = { gen_block : genesis[0] };
-    auto network = makeTestNetwork(conf);
+    auto network = makeTestNetwork(TestConf.init);
     network.start();
     scope(exit) network.shutdown();
     scope(failure) network.printLogs();
     network.waitForDiscovery();
 
+    auto gen_tx = GenesisTransaction();
```

Comment:
This is not the right genesis transaction. This is the main thread, and it doesn't have the custom genesis block that's set in `Base.d`. You should use `network.blocks`.

In fact all you have to do here is compare `node.getBlocksFrom(0, 1)[0]` with `network.blocks[0]`.

Reply:
So the `auto index = b.txs.enumerate.assumeSorted.find!(res => res[1] == gen_tx);` part should be replaced with just checking if the block is the same as the one in the APIManager (`network` variable).

Reply:
That's a good suggestion.